spec/solver_spec.rb

```diff
@@ -0,0 +1,31 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Solver methods should return correct result' do
+    before(:each) do
+      @solver = Solver.new
+    end
+
+    it 'should calculate factorial of ceratin input as integer number ' do
+      expect(@solver.factorial(2)).to eq 2
+      expect(@solver.factorial(5)).to eq 120
+      expect(@solver.factorial(10)).to eq 3_628_800
+      expect(@solver.factorial(15)).to eq 1_307_674_368_000
+    end
```

Comment:
- [ ] You have implemented brilliant processes here but you seem to be missing the need to test the code against unintended operations on negative numbers. Kindly write a test for such a negative case to raise exception error as contained in the requirement :point_down: 

![image](https://github.com/mohamedgamaleldin999999/TDD-ruby-practice/assets/99715640/305b2d52-340a-4533-a699-e69714c85d32)